client/src/app.js

```diff
@@ -40,6 +40,7 @@ class App extends Component {
     this.setState({ isAuthenticated: false }, () => {
       Auth.removeToken();
     });
+    localStorage.removeItem('userID');
```

Comment:
Very bad decision, your main class is responsible to remove information from localstorage. If we will change storage system, we 100% will forget to fix this line.
Your previous approach was well enough. You just needed to change method name